src/commands/emulators-exec.ts

```diff
@@ -12,12 +12,17 @@ import { DatabaseEmulator } from "../emulator/databaseEmulator";
 import { EmulatorRegistry } from "../emulator/registry";
 import { FirestoreEmulator } from "../emulator/firestoreEmulator";
 import * as commandUtils from "../emulator/commandUtils";
+import * as getProjectId from "../getProjectId";
 
-async function runScript(script: string): Promise<number> {
+async function runScript(script: string, projectId: string | undefined): Promise<number> {
```

Comment:
Something feels wrong about `runScript` taking one random extra argument specific to projectID.  Maybe it should take an `env: Map<String,String>` argument so we can pass random env to the script from the outside?

Reply:
Done.